src/Microsoft.DotNet.Wpf/src/PresentationUI/MS/Internal/documents/DocumentSignatureManager.cs

```diff
@@ -506,9 +506,7 @@ internal void ShowSigningDialog(IntPtr parentWindow)
                                             // Get[Algorithm]PrivateKey methods would always have returned the private key if the PrivateKey property would
                                             // But Get[Algorithm]PrivateKey methods never throw but returns null in case of error during cryptographic operations
                                             // But we want exception to be thrown when an error occurs during a cryptographic operation so that we can redisplay the certificate picker
-                                            #pragma warning disable SYSLIB0028
-                                            AsymmetricAlgorithm testKey = x509Certificate2.PrivateKey;
-                                            #pragma warning restore SYSLIB0028
+                                            AsymmetricAlgorithm testKey = x509Certificate2.GetRSAPrivateKey();
```

Comment:
@singhashish-wpf As per comments above, the caller is expecting exception, if cryptographic operations fails. GetRSAPrivateKey won't do that.
Please cross check the consequence of not having the exceptions.

Reply:
Throwing exception if the privatekey is null.
Commit : https://github.com/dotnet/wpf/pull/4968/commits/c43cd8d2b5f9e1f73c8122a06a074138cff430d0